tfx/components/transform/executor_test.py

```diff
@@ -58,9 +60,25 @@ class _TempPath(types.Artifact):
 # TODO(b/122478841): Add more detailed tests.
 class ExecutorTest(tft_unit.TransformTestCase):
 
-  def _get_source_data_dir(self):
-    return os.path.join(
+  @classmethod
+  def setUpClass(cls):
+    cls._temp_example_dir = tempfile.mkdtemp()
+    cls._source_data_dir = os.path.join(
         os.path.dirname(os.path.dirname(__file__)), 'testdata')
+    source_example_dir = os.path.join(cls._source_data_dir, 'csv_example_gen')
+
+    cls._artifact1_uri = os.path.join(cls._temp_example_dir, 'csv_example_gen1')
+    io_utils.copy_dir(source_example_dir, cls._artifact1_uri)
+    cls._artifact2_uri = os.path.join(cls._temp_example_dir, 'csv_example_gen2')
+    io_utils.copy_dir(source_example_dir, cls._artifact2_uri)
+
+    # Duplicate the number of train and eval records such that
+    # second artifact has twice as many as first.
+    artifact2_pattern = os.path.join(cls._artifact2_uri, '*', '*')
```

Comment:
Can this just be replaced with join(..., '**')?
(depends on the file structure, so I'm not sure)

Reply:
i think double-asterisk matches recursively (i.e. matches files at both 1 level depth and 2-level depth) when we only want to match files at the 2nd depth